.github/scripts/assemble-materials.py

```
@@ -0,0 +1,188 @@
+constructors = {}
+fbudget = 20_000
+for field in sorted(consumed_fields)[:12]:
+    if fbudget <= 0:
+        break
+    hits = []
+    raw = subprocess.run(['git', 'grep', '-n', '--no-color',
+                          '-E', re.escape(field) + r'[[:space:]]*=|^\s*' + re.escape(field) + r'[[:space:]]*:'],
+                         capture_output=True, text=True).stdout
+    # 优先 src/ 的构造点（真实管道），scripts/ 的测试/repro 假数据殿后——
+    # #551 盲测实证：字母序让 repro-*.tsx 的假 displayCwd 淹没 channel.ts 的真构造。
+    src_first = sorted((l for l in raw.splitlines()
```

Ruff (0.16.2)

[error] Ambiguous variable name: `l`

(E741)